internal/engine/compiler/engine.go

```diff
@@ -829,17 +843,18 @@ func (f *function) getSourceOffsetInWasmBinary(pc uint64) uint64 {
 	}
 }
 
-func NewEngine(_ context.Context, enabledFeatures api.CoreFeatures, fileCache filecache.Cache) wasm.Engine {
-	return newEngine(enabledFeatures, fileCache)
+func NewEngine(_ context.Context, enabledFeatures api.CoreFeatures, fileCache filecache.Cache, enabledPerfmap bool) wasm.Engine {
```

Comment:
Instead of passing a boolean parameter here, would it be possible to add a method like `EnablePerfMap(bool)`?

Reply:
Alternatively we could use a config struct for parameters, I'm concerned about seeing the number of arguments grow, we're setting a precedent that will encourage future contributions to keep adding parameters which will become unmanageable eventually.

Reply:
summary comment is that NewEngine can have a config struct like the other stuff. Seems getting to that spot and this is internal anyway, so worth a before/after (via spiking in a diff!)

Reply:
yes, I was considering the config struct but wanted to open the PR and get feedback before jumping on it. Will work on it!

Reply:
I wonder if it's worth spinning off the config struct in its own PR so that we get that in earlier, and with its own commit 🤔  wdyt?

Reply:
I'm personally fine with tools being introduced as they are used. While it makes the commit title harder to write, it is better overall I think.

longer..

Some projects create a lot of infra in PRs and sometimes forget to use that infra, and regardless it can be hard to figure out what was motivating something when they are disconnected. Basically, the slight mushing of work challenges folks to use what they write and also can reduce the amount of polish PRs if what they thought they needed wasn't exactly it. Besides.. it is internal anyway.

Reply:
@evacchi if this is mainly about time until merge "get that in earlier"

if you need it and this PR is still pending, just add one and I'm sure @Pryz won't mind rebasing. wazero has an implicit no hostage policy ;)

Reply:
oh no, I don't need it. I just wanted to let @Pryz move forward in smaller steps if there were any issues, but I have no objection in keeping this in the same PR